src/Compilers/VisualBasic/Portable/Symbols/SubstitutedNamedType.vb

```diff
@@ -31,6 +31,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             _substitution = substitution
         End Sub
 
+        Protected Shared Function WasConstructedForModifiers(type As NamedTypeSymbol) As Boolean
+            Do
+                Dim typeArguments = type.TypeArgumentsNoUseSiteDiagnostics
+                Dim typeParameters = type.OriginalDefinition.TypeParameters
+
+                For i = 0 To typeArguments.Length - 1
```

Comment:
>For i = 0 To typeArguments.Length - 1 [](start = 16, length = 37)

It feels like it would be better to iterate through the chain of TypeSubstitutions instead.